### PR TITLE
[Feat] Add PointCloud ObjectDetection

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/data_source.py [hunk 1]
```diff
@@ -176,6 +176,14 @@ def __hash__(self) -> int:
         return hash(self.value)
 
 
+class BaseDataFormat(LightningEnum):
+
+    pass
```
Comment (SeanNaren) [inline flash/core/data/data_source.py:181]: Don't need this `pass` do you?

Comment (tchaton) [reply flash/core/data/data_source.py]: Done !

**Review by SeanNaren — APPROVED** [review]
Don't know much about pointclouds, but LGTM :)

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: tests/pointcloud/detection/test_data.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+@pytest.mark.skipif(not _POINTCLOUD_TESTING, reason="pointcloud libraries aren't installed")
+def test_pointcloud_object_detection_data(tmpdir):
+
+    seed_everything(52)
+
+    download_data("https://pl-flash-data.s3.amazonaws.com/KITTI_micro.zip", "data")
+
+    dm = PointCloudObjectDetectorData.from_folders(train_folder=join("data", "KITTI_micro", "Kitti", "train"), )
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:37]:
```suggestion
    download_data("https://pl-flash-data.s3.amazonaws.com/KITTI_micro.zip", tmpdir)

    dm = PointCloudObjectDetectorData.from_folders(train_folder=join(tmpdir, "KITTI_micro", "Kitti", "train"), )
```

Comment (tchaton) [reply tests/pointcloud/detection/test_data.py]: This didn't work for some reason.

Comment (tchaton) [reply tests/pointcloud/detection/test_data.py]: > This didn't work for some reason.

File: tests/pointcloud/detection/test_data.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+@pytest.mark.skipif(not _POINTCLOUD_TESTING, reason="pointcloud libraries aren't installed")
+def test_pointcloud_object_detection_data(tmpdir):
+
+    seed_everything(52)
+
+    download_data("https://pl-flash-data.s3.amazonaws.com/KITTI_micro.zip", tmpdir)
+
+    dm = PointCloudObjectDetectorData.from_folders(train_folder=join(tmpdir, "KITTI_micro", "Kitti", "train"), )
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:37]:
```suggestion
    dm = PointCloudObjectDetectorData.from_folders(train_folder=join(tmpdir, "KITTI_Micro", "Kitti", "train"), )
```

File: tests/pointcloud/detection/test_data.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            assert batch.bboxes[0].shape == torch.Size([1, 7])
+            assert batch.attr[0]["name"] == '000000.bin'
+            assert batch.attr[1]["name"] == '000001.bin'
+
+    num_classes = 19
+    model = MockModel(backbone="pointpillars_kitti", num_classes=num_classes)
+    trainer = Trainer(max_epochs=1, limit_train_batches=1, limit_val_batches=0)
+    trainer.fit(model, dm)
+
+    predict_path = join("data", "KITTI_micro", "Kitti", "predict")
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:54]:
```suggestion
    predict_path = join("data", "KITTI_Micro", "Kitti", "predict")
```

File: tests/pointcloud/detection/test_data.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    download_data("https://pl-flash-data.s3.amazonaws.com/KITTI_micro.zip", tmpdir)
+
+    dm = PointCloudObjectDetectorData.from_folders(train_folder=join(tmpdir, "KITTI_Micro", "Kitti", "train"), )
+
+    class MockModel(PointCloudObjectDetector):
+
+        def training_step(self, batch, batch_idx: int):
+            assert isinstance(batch, ObjectDetectBatchCollator)
+            assert batch.point[0].shape == torch.Size([20730, 4])
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:43]:
```suggestion
            assert batch.point[0].shape == torch.Size([20188, 4])
```

File: tests/pointcloud/detection/test_data.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    download_data("https://pl-flash-data.s3.amazonaws.com/KITTI_micro.zip", tmpdir)
+
+    dm = PointCloudObjectDetectorData.from_folders(train_folder=join(tmpdir, "KITTI_Micro", "Kitti", "train"), )
+
+    class MockModel(PointCloudObjectDetector):
+
+        def training_step(self, batch, batch_idx: int):
+            assert isinstance(batch, ObjectDetectBatchCollator)
+            assert batch.point[0].shape == torch.Size([20188, 4])
+            assert batch.point[1].shape == torch.Size([18268, 4])
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:44]:
```suggestion
            assert batch.point[1].shape == torch.Size([18826, 4])
```

File: tests/pointcloud/detection/test_data.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    dm = PointCloudObjectDetectorData.from_folders(train_folder=join(tmpdir, "KITTI_Micro", "Kitti", "train"), )
+
+    class MockModel(PointCloudObjectDetector):
+
+        def training_step(self, batch, batch_idx: int):
+            assert isinstance(batch, ObjectDetectBatchCollator)
+            assert batch.point[0].shape == torch.Size([20188, 4])
+            assert batch.point[1].shape == torch.Size([18826, 4])
+            assert batch.bboxes[0].shape == torch.Size([1, 7])
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:45]:
```suggestion
            assert batch.bboxes[0].shape == torch.Size([7, 7])
```

File: tests/pointcloud/detection/test_data.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            assert batch.bboxes[0].shape == torch.Size([7, 7])
+            assert batch.attr[0]["name"] == '000000.bin'
+            assert batch.attr[1]["name"] == '000001.bin'
+
+    num_classes = 19
+    model = MockModel(backbone="pointpillars_kitti", num_classes=num_classes)
+    trainer = Trainer(max_epochs=1, limit_train_batches=1, limit_val_batches=0)
+    trainer.fit(model, dm)
+
+    predict_path = join("data", "KITTI_Micro", "Kitti", "predict")
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:54]:
```suggestion
    predict_path = join(tmpdir, "KITTI_Micro", "Kitti", "predict")
```

File: tests/pointcloud/detection/test_data.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    trainer = Trainer(max_epochs=1, limit_train_batches=1, limit_val_batches=0)
+    trainer.fit(model, dm)
+
+    predict_path = join(tmpdir, "KITTI_Micro", "Kitti", "predict")
+    model.eval()
+
+    predictions = model.predict([join(predict_path, "scans/000000.bin")])
+    assert torch.stack(predictions[0][DefaultDataKeys.INPUT]).shape[1] == 4
+    assert len(predictions[0][DefaultDataKeys.PREDS]) == 158
+    assert predictions[0][DefaultDataKeys.PREDS][0].__dict__["identifier"] == 'box:9'
```
Comment (ethanwharris) [inline tests/pointcloud/detection/test_data.py:60]:
```suggestion
    assert predictions[0][DefaultDataKeys.PREDS][0].__dict__["identifier"] == 'box:1'
```

**Pull request merged**